plugins/module_utils/s3.py

```diff
@@ -151,3 +151,26 @@ def s3_extra_params(options, sigv4=False):
         config["signature_version"] = "s3v4"
     extra_params["config"] = config
     return extra_params
+
+
+# _list_backup_inventory_configurations is a workaround for a missing paginator for listing
+# bucket inventory configuration in boto3:
+# https://github.com/boto/botocore/blob/1.34.141/botocore/data/s3/2006-03-01/paginators-1.json
+def _list_bucket_inventory_configurations(client, bucket_name):
+    first_iteration = False
+    next_token = None
+
+    response = client.list_bucket_inventory_configurations(Bucket=bucket_name)
+    next_token = response.get("NextToken", None)
+
+    if next_token is None:
+        return response.get("InventoryConfigurationList", [])
+
+    entries = []
+    while next_token is not None:
+        if first_iteration:
+            response = client.list_bucket_inventory_configurations(NextToken=next_token)
```

Comment:
```suggestion
            response = client.list_bucket_inventory_configurations(Bucket=bucket_name, NextToken=next_token)
```

With pagination you generally need to pass both the NextToken **and** the other parameters